### ISSUE TITLE
Ensure rancher-server chart works with k8s because of removed API in k8s 1.20

### ISSUE THREAD

**Issue by cjellick** [rancher/rancher#22487]
https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/

As of K8's 1.20 the ingress api endpoint will be fully migrated away from extensions.

Our rancher-server chart will need to be updated to handle that: https://github.com/rancher/rancher/blob/master/chart/templates/ingress.yaml#L1-L2

We will also need to ensure our chart works with 1.18 in general.

**Comment by cbron** [comment]
Repurposing ticket.

The title of the blog post and body don't match. Ingress doesn't get dropped until 1.18, not 1.16: 
> Ingress: will no longer be served from extensions/v1beta1 in **v1.18.**

Also we can't migrate anyway until we drop support for < 1.14.

> Migrate to the networking.k8s.io/v1beta1 API, serving Ingress since v1.14

**Comment by cbron** [comment]
Change is basically just this: https://github.com/cbron/rancher/commit/8f90d9168aef574856e585c09b30061e86a73915

**Comment by cbron** [comment]
This one is also fixed by https://github.com/rancher/rancher/issues/29144
There is another ticket linked there to handle moving from `networking/v1beta1` to `networking/v1`, which is the next step after moving away from the extensions path.

**Comment by jiaqiluo** [comment]
The validation is done as part of #29144 in this comment https://github.com/rancher/rancher/issues/29144#issuecomment-742066349

TLDR:  the ingress will be created with either the apiVersion `networking.k8s.io/v1` or `networking.k8s.io/v1beta1` based on the capability of the cluster.
